contrib/babelfishpg_tsql/src/collationproperty.c

```diff
@@ -19,17 +19,29 @@
 PG_FUNCTION_INFO_V1(collationproperty);
 
 extern bytea *convertIntToSQLVariantByteA(int ret);
+extern bytea *convertBinaryToSQLVariantByteA(int64_t ret);
+
 extern  coll_info_t coll_infos[];
 
 Datum collationproperty(PG_FUNCTION_ARGS)
 {
 	const char *collationname = text_to_cstring(PG_GETARG_TEXT_P(0));
 	const char *property = text_to_cstring(PG_GETARG_TEXT_P(1));
-	int result = -1;
+	int64_t result64 = -1;
+	int result32 = -1;
+
+	if (strcasecmp(property, "tdscollation") == 0)
+		result64 = tsql_tdscollationproperty_helper(collationname, property);
+  	else
+  		result32 = tsql_collationproperty_helper(collationname, property);
 
-	result = tsql_collationproperty_helper(collationname, property);
 
-	if (result != -1)
-		PG_RETURN_BYTEA_P(convertIntToSQLVariantByteA(result));
+	if (result32 != -1)
+		PG_RETURN_BYTEA_P(convertIntToSQLVariantByteA(result32));
+	if (result64 != -1)
+	{
+		/* SQL variant Base type for tdscollation is Binary, while for the rest is INT. */
+		PG_RETURN_BYTEA_P(convertBinaryToSQLVariantByteA(result64));
+	}
```

Comment:
This can be simplified in single if... else...
if (property == 'tdscollation')
{
 res = tsql_tdscollationproperty_helper(collationname, property);
 if (res != -1)
     PG_RETURN_BYTEA_P(convertBinaryToSQLVariantByteA(result64));
}
else
{
 res = tsql_collationproperty_helper(collationname, property);
 if (res != -1)
     PG_RETURN_BYTEA_P(convertIntToSQLVariantByteA(result32));
}

Reply:
Implemented the logic